### ISSUE TITLE
feature request - rolling version update for apps

### ISSUE THREAD

**Issue by PierreKircher** [deis/deis#1829]
at some point you need to test your app with real traffic 

you got v5 deployed and its stable now works ok and now you need to create the next killer feature .. 

feature ready .. testing done .. (but nothing tests as good as real traffic .. !)

.. v6 .. but to see if anything behaves correctly you need to have live traffic ..

100% direct traffic could cost you a lot in a given scenario .. 
of course you can rollback but you still affect 100% of your user-base 

so a rolling update function 
for balancing would be nice .. 
lets say for the next N hours  (timescale should be "adjustable" i am going to use %)  gradually update traffic towards v6 

why ? simple ..  risks management ( cost of fail )
                V5       V6
10%|        90        10     - ( get some peaks with a few customers .. - easy to spot errors ..)
20%|        80        20     - 
30%|        70        30     - 
40%|        60        40     - 
50%|        50        50     - ( decent amount of traffic get realistic insight )
60%|        40        60     - 
70%|        30        70     - 
80%|        20        80     - ( nearly there .. if anything goes wrong we should have spotted it by now )
90%|        10        90     - 
100 full deploy of v6 kill - v5 .. ( we are able to restore it anyway ) 

if something errors in that timescale .. you could simply rollback to 100% v5 .. fix your problem and deploy v5 to v7 with the same mechanics ..

HOW ?

we use nginx as routing mechanics anyway ( as of now ).. 

weight=number
    sets the weight of the server, by default, 1. 

 upstream XXX {
        server v5.example.com weight=9;
        server v6.example.com weight=1;
}

basically what we need is a way to tell to not kill the older versions and keep it around .. to verify

deis config:set GRADUAL_DEPLOY=Xm/h

any ideas / insights would be appreciated

**Comment by jwaldrip** [comment]
:+ plus one billion. Seriously... this should be expected for a production scale application.

**Comment by mboersma** [comment]
Closing in favor of #3685.
